mne/annotations.py

```diff
@@ -1328,28 +1454,51 @@ def _read_annotations_txt_parse_header(fname):
     def is_orig_time(x):
         return x.startswith("# orig_time :")
 
+    def is_columns(x):
+        return x.startswith("# onset, duration, description")
+
     with open(fname) as fid:
         header = list(takewhile(lambda x: x.startswith("#"), fid))
 
     orig_values = [h[13:].strip() for h in header if is_orig_time(h)]
     orig_values = [_handle_meas_date(orig) for orig in orig_values if _is_iso8601(orig)]
 
-    return None if not orig_values else orig_values[0]
+    columns = [[c.strip() for c in h[2:].split(",")] for h in header if is_columns(h)]
+
+    return None if not orig_values else orig_values[0], (
+        None if not columns else columns[0]
+    )
 
 
 def _read_annotations_txt(fname):
     with warnings.catch_warnings(record=True):
         warnings.simplefilter("ignore")
         out = np.loadtxt(fname, delimiter=",", dtype=np.bytes_, unpack=True)
-    ch_names = None
+    orig_time, columns = _read_annotations_txt_parse_header(fname)
+    ch_names = extras = None
     if len(out) == 0:
         onset, duration, desc = [], [], []
     else:
-        _check_option("text header", len(out), (3, 4))
-        if len(out) == 3:
-            onset, duration, desc = out
+        if columns is None:
+            _check_option("text header", len(out), (3, 4))
+            columns = ["onset", "duration", "description"] + (
+                ["ch_names"] if len(out) == 4 else []
+            )
         else:
-            onset, duration, desc, ch_names = out
+            _check_option(
+                "text header", columns[:3], (["onset", "duration", "description"],)
+            )
+            _check_option("text header len", len(out), (len(columns),))
+        onset, duration, desc = out[:3]
+        i_col = 3
+        if len(columns) > i_col and columns[i_col] == "ch_names":
+            ch_names = out[i_col]
+            i_col += 1
+        if len(columns) > i_col:
+            extras = [
+                {columns[j_col]: out[j_col][i] for j_col in range(i_col, len(columns))}
+                for i in range(len(onset))
+            ]
```

Comment:
this section here feels a bit hard to maintain. Can you take a few minutes to see if you can simplify the logic or otherwise reduce the complexity?  If you're out of ideas LMK and I can push something.